### PR TITLE
Navigation component: Add color to menu and group title

### PR DESCRIPTION
**Pull request by david-szabo97**
## Description
Add `color` property to `MenuTitleUi` and `GroupTitleUi` to make them more specific.

See https://github.com/WordPress/gutenberg/pull/25506

## Types of changes
Bug Fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by Copons** [comment]
@david-szabo97 I'd suggest adding these colors to the original PR #25506 instead, and close this.

At first I was about to ask you to replace the hex codes with the [G2 color](https://github.com/WordPress/gutenberg/blob/f13d572f71a113222154b6d39efcf3f39099786e/packages/components/src/utils/colors-values.js#L35), but then I realized we already use that same hex color (and others) elsewhere in the Nav styles.

Eventually we'll need to discuss, revamp, and polish the menu colors, so imho this change is not essential, even more so as #25506 is currently an exploration. 🙂

File: packages/components/src/navigation/styles/navigation-styles.js [hunk 1]
```diff
@@ -50,12 +50,14 @@ export const MenuBackButtonUI = styled( Button )`
 export const MenuTitleUI = styled( Text )`
 	padding: 4px 0 4px 16px;
 	margin-bottom: 8px;
+	color: #f0f0f0;
```
Comment (vindl) [inline packages/components/src/navigation/styles/navigation-styles.js:53]: Could we use `G2.gray[100]` instead of hardcoding the hex value?

**Comment by david-szabo97** [comment]
@Copons Eventually we will need to add a color to these elements. Since they aren't a simple div or span which is usually unstyled. I'm fine with adding this little change in the exploration PR, but I'd rather have it merged to make sure we don't forget about it. I'm sure the color will change in the future when the design is finalized, but it's easier to find and replace than figure out why it isn't working as expected.

File: packages/components/src/navigation/styles/navigation-styles.js (hunk 1 above)
Comment (david-szabo97) [reply packages/components/src/navigation/styles/navigation-styles.js]: Replaced it with G2

**Review by vindl — APPROVED** [review]

**Review by Copons — APPROVED** [review]

**Pull request merged**